aten/src/THC/THCApply.cuh

```diff
@@ -236,7 +240,7 @@ bool THC_pointwiseApply1(THCState* state,
   kernelPointwiseApply1<Op,                                             \
                         typename TensorUtils<TensorTypeA>::DataType,    \
                         TYPE, A>                                        \
-    <<<grid, block, 0, THCState_getCurrentStream(state)>>>(             \
+    <<<grid, block, 0, THCState_getCurrentStreamOnDevice(state, curDevice)>>>(             \
```

Comment:
You're never checking that `curDevice` is valid (it wasn't an error)

Reply:
You will never get here if there was an error, getApplyGrid will return an error.

Reply:
Hmm ok. Making error checking part of `getApplyGrid`'s contract is quite weird 😕 Can we add an `AT_CHECK` there? It's pretty much free.